src/test/java/core/mate/academy/service/MachineServiceTest.java

```diff
@@ -23,28 +23,28 @@ public static void beforeClass() {
 
     @Test
     public void getAll_bulldozerType() {
-        List<Machine> actual = machineService.getAll(Bulldozer.class);
+        List<? extends Machine> actual = machineService.getAll(Bulldozer.class);
         assertEquals("The list should contain elements of type Bulldozer",
             Bulldozer.class, actual.get(0).getClass());
     }
 
     @Test
     public void getAll_excavatorType() {
-        List<Machine> actual = machineService.getAll(Excavator.class);
+        List<? extends Machine> actual = machineService.getAll(Excavator.class);
         assertEquals("The list should contain elements of type Excavator",
             Excavator.class, actual.get(0).getClass());
     }
 
     @Test
     public void getAll_truckType() {
-        List<Machine> actual = machineService.getAll(Truck.class);
+        List<? extends Machine> actual = machineService.getAll(Truck.class);
         assertEquals("The list should contain elements of type Truck",
             Truck.class, actual.get(0).getClass());
     }
 
     @Test
     public void getAll_invalidType() {
-        List<Machine> actual = machineService.getAll(Machine.class);
+        List<? extends Machine> actual = machineService.getAll(Machine.class);
         assertEquals("For the unknown class you should return empty list",
             Collections.emptyList(), actual);
```

Comment:
This test may pass incorrectly because `Collections.emptyList()` is immutable and cannot be compared directly with a mutable list that `getAll` might return. You should instead check that the returned list is empty using `assertTrue(actual.isEmpty())`.